packages/execution-environments/src/common/endowments/network.ts

```diff
@@ -62,7 +143,10 @@ const createNetwork = () => {
       };
       openConnections.add(openFetchConnection);
 
-      res = await fetchPromise;
+      res = new ResponseWrapper(
+        await withTeardown(fetchPromise, teardownRef),
+        teardownRef,
+      ) as unknown as Response;
```

Comment:
Why do you need the cast to `unknown` here? If it implements `Response`, duck typing should work.

Reply:
Probably added when it was partially implemented. Removed it now.